dev-server/ui-playground/components/PatternPage.js

```diff
@@ -0,0 +1,140 @@
+import { toChildArray } from 'preact';
+
+import { jsxToString } from '../util/jsx-to-string';
+
+/**
+ * Components to render patterns and examples. Typical structure of a "Demo"
+ * page might be:
+ *
+ * <PatternPage title="Zoo Animals">
+ *   <Pattern title="Elephant">
+ *     <p>Everyone loves an elephant.</p>
+ *
+ *     <PatternExamples title="Colored elephants">
+ *       <PatternExample details="Pink elephant">
+ *          <Elephant color="pink" />
+ *       </PatternExample>
+ *       <PatternExample details="Blue elephant">
+ *          <Elephant color="blue" />
+ *       </PatternExample>
+ *     </PatternExamples>
+ *
+ *     <PatternExamples title="Patterned elephants">
+ *       <PatternExample details="Spotted elephant">
+ *          <Elephant variant="spotted" />
+ *       </PatternExample>
+ *     </PatternExamples>
+ *   </Pattern>
+ *
+ *   <Pattern title="Monkeys">
+ *     <PatternExamples>
+ *       <PatternExample details="Your basic monkey">
+ *         <Monkey />
+ *       </PatternExample>
+ *     </PatternExamples>
+ *   </Pattern>
+ * </PatternPage>
+ */
+
+/**
+ * @typedef PatternPageProps
+ * @prop {import("preact").ComponentChildren} children
+ * @prop {string} title
+ */
+
+/**
+ * Render a page of patterns
+ * @param {PatternPageProps} props
```

Comment:
Just as a stylistic convention, we normally leave a blank line between the summary and `@param` lines.